lib/ffi.rb

```diff
@@ -8,10 +8,14 @@
 
   require 'ffi/ffi'
 
-elsif RUBY_ENGINE == 'jruby' && Gem::Version.new(JRUBY_VERSION) >= Gem::Version.new("9.3.pre")
+elsif RUBY_ENGINE == 'jruby' && Gem::Version.new(RUBY_ENGINE_VERSION) >= Gem::Version.new("9.3.pre")
   JRuby::Util.load_ext("org.jruby.ext.ffi.FFIService")
   require 'ffi/ffi'
 
+elsif RUBY_ENGINE == 'truffleruby' && Gem::Version.new(RUBY_ENGINE_VERSION) >= Gem::Version.new("20.1.0")
```

Comment:
We could tweak this condition to `>= Gem::Version.new("20.1.0-dev-a")` once the related changes are merged in TruffleRuby (nightly versions look like `20.1.0-dev-374d84d2`).
That way the CI running on truffleruby-head would return `true` for this condition.
(Potentially we could use `RUBY_RELEASE_DATE` too)

Reply:
I've done this, the latest TruffleRuby nightly now has the changes needed.